File: Yow.CoD.Finance.Domain/Contracts/LoanCreatedEvent.cs
using System;

namespace Yow.CoD.Finance.Domain.Contracts
{
    public sealed class LoanCreatedEvent : Event
    {
        public LoanCreatedEvent(DateTimeOffset createdOn, decimal amount, Duration term, PaymentPlan paymentPlan)
        {
            CreatedOn = createdOn;
            Amount = amount;
            Term = term;
            PaymentPlan = paymentPlan;
        }

        public DateTimeOffset CreatedOn { get; }        
        public decimal Amount { get; }
        public Duration Term { get; }
        public PaymentPlan PaymentPlan { get; }
    }
}

File: java/domain/src/main/java/com/leecampbell/cod/domain/contracts/LoanCreatedEvent.java
package com.leecampbell.cod.domain.contracts;

import java.math.BigDecimal;
import java.time.OffsetDateTime;

public final class LoanCreatedEvent extends DomainEvent {
    private final OffsetDateTime createdOn;
    private final Duration term;
    private final PaymentPlan paymentPlan;
    private final BigDecimal amount;

    public LoanCreatedEvent(OffsetDateTime createdOn, Duration term, PaymentPlan paymentPlan, BigDecimal amount) {
        this.createdOn = createdOn;
        this.term = term;
        this.paymentPlan = paymentPlan;
        this.amount = amount;
    }

    public OffsetDateTime getCreatedOn() {
        return createdOn;
    }

    public Duration getTerm() {
        return term;
    }

    public PaymentPlan getPaymentPlan(){
        return paymentPlan;
    }

    public BigDecimal getAmount() {
        return amount;
    }
}